src/platform/forms-system/src/js/widgets/TextWidget.jsx

```diff
@@ -2,13 +2,24 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 const numberTypes = new Set(['number', 'integer']);
+const indexRegex = /:index/g;
 
 export default function TextWidget(props) {
+  const pageIndex = props.formContext?.pagePerItemIndex;
   let inputType = props.options.inputType;
   if (!inputType) {
     inputType = numberTypes.has(props.schema.type) ? 'number' : props.type;
   }
 
+  /**
+   * replaceIndex
+   * ariaDescribedby id may need to be indexed, add `:index` to the id and the
+   * formContext pagePerItemIndex will replace it
+   * @param {String|null} id - aria-describedby id with ":index" placeholder
+   */
+  const replaceIndex = (id = '') =>
+    id && pageIndex !== 'undefined' ? id.replace(indexRegex, pageIndex) : id;
```

Comment:
I'm not sure I like this, because of having to add `:index` into a string. Could we just append the index?

Reply:
That would work, but we'd have to be aware that it needs to match the ID of the hint text (dashes and all). I added a comment in the `TextWidget`